### PR TITLE
feat: setup end to end

### PR DESCRIPTION
**Pull request by GustaveCharles**
### What I did

- Setup a framework for end to end tests. 
- Setup EAS build environment.

### How I did it

I used the new Maestro framework. You can find the docs with [this link ](https://maestro.mobile.dev/ ). To make Maestro work you need to have an emulator. For this PR and for our actual eas.json, only an Android emulator works. You can setup Expo Orbit to manage emulators on [this link](https://docs.expo.dev/build/orbit/)

### What it looks like

https://github.com/uniconnect-epfl/uniconnect/assets/91184289/c28448a0-05e9-4bf9-a02e-98467b8ecf64

### CODE REVIEW

**Comment by alexandre-mrt** [comment]
LGTM.

**Comment by alexandre-mrt** [comment]
LGTM

File: uni-connect-v0.apk
Comment (Aidasvenc) [inline uni-connect-v0.apk:1]: I don't think we should push this file into main

**Review by Aidasvenc — CHANGES_REQUESTED** [review]
Everything looks good except that I think the .apk itself is not needed

File: uni-connect-v0.apk
Comment (Aidasvenc) [reply uni-connect-v0.apk:1]: Refactored!

**Review by Aidasvenc — APPROVED** [review]
LGTM now!

**Pull request merged**